packages/mobile/src/analytics/ValoraAnalytics.ts

```diff
@@ -0,0 +1,153 @@
+import Analytics, { Analytics as analytics } from '@segment/analytics-react-native'
+import Firebase from '@segment/analytics-react-native-firebase'
+import DeviceInfo from 'react-native-device-info'
+import { AppEvents } from 'src/analytics/Events'
+import { AnalyticsPropertiesList } from 'src/analytics/Properties'
+import { SEGMENT_API_KEY } from 'src/config'
+import { store } from 'src/redux/store'
+import Logger from 'src/utils/Logger'
+import * as Web3Utils from 'web3-utils'
+
+const TAG = 'ValoraAnalytics'
+
+async function getDeviceInfo() {
+  return {
+    AppName: DeviceInfo.getApplicationName(),
+    Brand: DeviceInfo.getBrand(),
+    BuildNumber: DeviceInfo.getBuildNumber(),
+    BundleId: DeviceInfo.getBundleId(),
+    Carrier: await DeviceInfo.getCarrier(),
+    DeviceId: DeviceInfo.getDeviceId(),
+    FirstInstallTime: await DeviceInfo.getFirstInstallTime(),
+    FontScale: await DeviceInfo.getFontScale(),
+    FreeDiskStorage: await DeviceInfo.getFreeDiskStorage(),
+    InstallReferrer: await DeviceInfo.getInstallReferrer(),
+    InstanceID: await DeviceInfo.getInstanceId(),
+    LastUpdateTime: await DeviceInfo.getLastUpdateTime(),
+    Manufacturer: await DeviceInfo.getManufacturer(),
+    MaxMemory: await DeviceInfo.getMaxMemory(),
+    Model: DeviceInfo.getModel(),
+    ReadableVersion: DeviceInfo.getReadableVersion(),
+    SerialNumber: await DeviceInfo.getSerialNumber(),
+    SystemName: DeviceInfo.getSystemName(),
+    SystemVersion: DeviceInfo.getSystemVersion(),
+    TotalDiskCapacity: await DeviceInfo.getTotalDiskCapacity(),
+    TotalMemory: await DeviceInfo.getTotalMemory(),
+    UniqueID: DeviceInfo.getUniqueId(),
+    UserAgent: await DeviceInfo.getUserAgent(),
+    Version: DeviceInfo.getVersion(),
+    isEmulator: await DeviceInfo.isEmulator(),
+    isTablet: DeviceInfo.isTablet(),
+  }
+}
+
+const SEGMENT_OPTIONS: analytics.Configuration = {
+  using: [Firebase],
+  flushAt: 20,
+  trackAttributionData: false,
+  debug: __DEV__,
+  trackAppLifecycleEvents: true,
+  recordScreenViews: true,
+  ios: {
+    trackAdvertising: false,
+    trackDeepLinks: true,
+  },
+}
+
+class ValoraAnalytics {
+  sessionId: string
+  userAddress: string | null | undefined
+  deviceInfo: object
+
+  constructor() {
+    this.sessionId = ''
```

Comment:
Nit, looks like we don't even need the constructor anymore at this point, you can just assign these defaults above.